### DISCUSSION TITLE
No idea what I'm doing

### DISCUSSION THREAD

**Discussion by ArcusConsulting** [smartcontractkit/full-blockchain-solidity-course-py#456] [General]
I'm about 6h30 in to the video, which has taken me almost 2 months to get through (juggling full time job, consulting biz, selling house & buying another one, and studying for certifications). I wish I was able to dedicate more time per day to this... Since I can only allocate 30-60min of real time, in which I only get through 15-30min of actual content, I lose track of what I'm trying to accomplish with this video. in other words, I have no idea what these programs and scripts are supposed to be doing. I don't see how I can integrate this in any real-life scenario or how I could charge other people/businesses/companies for writing these smart contracts.

All that to say, I do believe Web3 and blockchain are the way of the future so I'll be damned if I don't finish this 16h video (I have some vacation in December so I hope to make up ground then). I've attempted countless times over the last 5 years to learn programming... any kind of programming... to no avail. I tried Python 4 years ago, gave up after a few months because I found the programming space too overwhelming. Web design didn't interest me cause I'm not creative; AI & ML are in a different galaxy as far as learning curve goes, I didn't understand the marketability of we scraping or data science, and I don't have any ideas for new apps. All the apps I need exist! I have zero interest in making Twitter or Instagram clones (seems like waste of time re-creating something that already exists), and I'm a mediocre writer at best, so blogging about my programming journey was also not an option. Point is, every now and then I'll fool myself into thinking I'll learn how to code or program, and I find myself 4 years later in my journey, not having written or come of up with a single original line of code that I can lay claim to. I'm really good with SQL (need it for my job), but I've come to understand it is a fallacy to call SQL a programming language. And I haven't been able to apply any of my SQL skills to other programming languages.

It seems like every time I think I'm making progress, there is something else I need to learn... Like taking 1 step forward and 5 steps back. Data structures... algorithms... math... Want to build a website? Learn HTML, oh and CSS! Oh and you have to pick a stack... Python? Better learn Django, too! Then you have to learn Git... and GitHub... and how to host your website... And if you don't want to pay $5,000 for a website, you have to learn web design, unless you want to use very bland HTML templates...

But there's something about being part of a movement that has yet to really explode... To be at ground zero. I haven't quite figured out yet what that feeling is. I still have no idea how to monetize any of what I'm learning with blockchain. Maybe I'm just not adept at learning how to program, just like how some people will never be able to play guitar, no matter how much they practice.

I want to be part of something exciting! And I really think this is it. Blockchain is the way forward. I recently obtained by CBP (Certified Bitcoin Professional) certification from C4, but it isn't a technical certification (zero coding involved). I'm fascinated with the space and continue to read and educate myself everyday... But godamn programming is difficult. I'm amazed at how @PatrickAlphaC can just effortlessly write these beautiful programs, and know how all the scripts map together and work harmoniously, and be aware of all the required dependencies and how they all interact with eachother, and know what to import in which script, and write test scripts that test your actual scripts but it's in a test environment... It all blows my mind how someone has such an expert-level grasp on this technology. It's like child's play for him. 

I come from an accounting background, I'm turning 30, and only the last 5 years have I started to become aware of disruptive technologies. Also around the same time I started to shift how I think and really focus on 5, 10, 15 and 20 years into the future, and try to figure out where technology is headed, which new industries will thrive and which will fall by the wayside. I trylu believe those who can look 20 years into the future are the ones making their fortunes (the Elon Musk's and the Jeff Bezos' of the world). I've never been a forward or long-term thinker in that sense. This shift in perspective has affected how I invest my money, the books I read, what I choose to learn... I've never seen any of my feeble attempts at programming all the way till the end, but maybe that's the issue. Learning to program, or learning for that matter, never really ends. I think I just need to pick 1 lane and stay in it, and I think that lane is blockchain.

That's enough for today... I'm hoping to get to hour 7 of 16 by EOD tomorrow. 

Hope everyone had a great weekend, and that you all have a great week!

Happy learning everyone,

- Eric

**Comment by cromewar** [comment]
Hello @ArcusConsulting, wow thanks for sharing your story here, I know how overwhelming it could be, all the new technologies and not knowing where exactly to go, what exactly to build or what to learn. I felt myself pretty much reflected on some of the points you mentioned. I had learned how to code some years ago, and I was on the same path of learning something, and then releasing there was other thing new, and when I finally thought I learned enough there were a thousand new things to learn.

There are some things I learned on the path, first of all you literally don't need to know anything to do things, lets suppose you want to create a web app using react, JS, you don't need to take a course about react, and trying to learn again the same basic programming syntax you already know or learned on another course, if you want to anything, just do it. On this case the freecodecamp course is just like a reference, if you want to create smart contracts just do it, everything is on Google.

That brings me to the second point, and that is, everyone and I mean literally everyone copies code and use someone else project as reference, so don't be frustrated if you don't understand everything, or how packages interact with each other and so on, you'll get experience by making mistakes, get stuck and building things, if you want to learn just build projects, and I understand you don't want to create a instagram clone so try to build utility things you would actually use, like a ToDo list app or something like that.

I really hope you find your way to learn and apply the new technologies on your career, and if blockchain is your goal to go, the freecodecamp course, this github repo and discord communities are the best point to start.
